app/frontend/src/components/EditLocationMap.tsx

```diff
@@ -55,136 +74,251 @@ export default function EditLocationMap({
+    map.current!.on("touchstart", "circle", onCircleTouch);
 
     const canvas = map.current!.getCanvas();
     const setCursorMove = () => (canvas.style.cursor = "move");
     const unsetCursor = () => (canvas.style.cursor = "");
-    map.current!.on("mouseenter", "location", setCursorMove);
-    map.current!.on("mouseleave", "location", unsetCursor);
+    map.current!.on("mouseenter", "circle", setCursorMove);
+    map.current!.on("mouseleave", "circle", unsetCursor);
   };
 
   const flyToSearch = (location: LngLat) => {
     map.current!.flyTo({ center: location, zoom: 13 });
-    onUp(
-      {
-        lngLat: location,
-      } as MapMouseEvent,
-      () => null
-    );
+    if (!exact) {
+      const randomizedLocation = displaceLngLat(
+        location,
+        Math.random() * radius.current,
+        Math.random() * 2 * Math.PI
+      );
+      onCircleUp(
+        {
+          lngLat: randomizedLocation,
+        } as MapMouseEvent,
+        () => null
+      );
+    } else {
+      onCircleUp(
+        {
+          lngLat: location,
+        } as MapMouseEvent,
+        () => null
+      );
+    }
   };
 
   return (
     <>
-      {error && <Alert severity="error">{error}</Alert>}
-      <Box
+      <div
         className={classNames(
           classes.root,
           { [classes.grow]: grow },
           className
         )}
       >
-        <Map
-          initialZoom={location ? 13 : 2}
-          initialCenter={centerCoords.current!}
-          postMapInitialize={initializeMap}
-          grow
-          {...otherProps}
-        />
-        <MapSearch
-          value={address || ""}
-          setError={setError}
-          setValue={setAddress}
-          setMarker={flyToSearch}
+        {error && <Alert severity="error">{error}</Alert>}
+        <div className={classNames(classes.map)}>
+          <Map
+            initialZoom={location ? 13 : 0.5}
+            initialCenter={centerCoords.current!}
+            postMapInitialize={initializeMap}
+            grow
+            {...otherProps}
+          />
+          <MapSearch
+            setError={setError}
+            setAddress={(_, simplified) =>
+              commit((rLocation) => {
+                return { ...rLocation, address: simplified };
+              })
+            }
+            setMarker={flyToSearch}
+          />
+        </div>
+        <div className={classes.coordinateBox}>
+          ({rLocation.lat.toFixed(5)}, {rLocation.lng.toFixed(5)})
+        </div>
+        {!hideRadiusSlider && (
+          <>
+            <Typography id="location-radius" gutterBottom>
+              Location accuracy
+            </Typography>
+            <Slider
+              aria-labelledby="location-radius"
+              value={rLocation.radius}
+              min={userLocationMinRadius}
+              max={userLocationMaxRadius}
+              onChange={(_, value) => {
+                radius.current = value as number;
```

Comment:
very disappointed in MUI that this is necessary :|

Reply:
it's returned as `number | number[]` because of range sliders or whatever 🤷‍♂️